web/src/giphy.ts

```diff
@@ -44,11 +50,11 @@ export function update_giphy_rating() {
     }
 }
 
-function get_rating() {
+function get_rating(): Rating {
     const options = realm.giphy_rating_options;
-    for (const rating in realm.giphy_rating_options) {
-        if (options[rating].id === realm.realm_giphy_rating) {
-            return rating;
+    for (const [rating, rating_value] of Object.entries(options)) {
+        if (rating_value.id === realm.realm_giphy_rating) {
+            return rating as Rating;
```

Comment:
This is a significant issue, which i tried solving but was unable to. The return type is `Rating `and it is not very easy to convince TypeScript that the keys of options is of Type `Rating` & disabled and not String. How can we address the issue here ?

Reply:
One option would be to have the client parse `giphy_rating_options` into that type using Zod.